mypy/checkexpr.py

```diff
@@ -3937,21 +3954,34 @@ def is_valid_var_arg(self, typ: Type) -> bool:
 
     def is_valid_keyword_var_arg(self, typ: Type) -> bool:
         """Is a type valid as a **kwargs argument?"""
+
         if self.chk.options.python_version[0] >= 3:
-            return is_subtype(typ, self.chk.named_generic_type(
-                'typing.Mapping', [self.named_type('builtins.str'),
-                                   AnyType(TypeOfAny.special_form)]))
+            return (
+                (is_subtype(typ, self.chk.named_generic_type(
+                    'typing.Mapping',
+                    [self.named_type('builtins.str'),
+                     AnyType(TypeOfAny.special_form)])))
+                or
+                (is_subtype(typ, self.chk.named_type(
+                    'typing.Mapping')) and
+                    typ.args and  # type: ignore
```

Comment:
Similar to above, you'll need to use `map_instance_to_supertype`. Also, ensure that `typ` is an instance of `Instance`, and you can get rid of the `# type: ignore`. Right now this is unsafe, since TypedDict types can be subtypes of `Mapping` but they don't have `args`.